src/main/java/Gumball/TaskList.java

```diff
@@ -0,0 +1,147 @@
+package Gumball;
+
+import java.util.ArrayList;
+
+public class TaskList {
+    public ArrayList<Task> tasks;
+    public int n = 0;
+
+    public TaskList() {
+        tasks = new ArrayList<Task>();
+    }
+
+    /**
+     * Returns String which provides information on Task being added.
+     * @param adding The Task being added to the TaskList.
+     * @return information of the Task being added.
+     * @throws InputErrorException
+     */
+    public String add(Task adding) throws InputErrorException {
+        tasks.add(adding);
+        n++;
+        return getSpecific(n);
+    }
+
+    /**
+     * Returns number of tasks in list.
+     * @return number of tasks in list.
+     */
+    public int getN() {
```

Comment:
Perhaps could consider using a more descriptive name like `getSize` as it is not immediately obvious to the caller what N here might refer to (and neither is it mentioned in the comment)